### PR TITLE
cvat formatting

### LINKED ISSUE #7: CVAT pipeline
Once the query is ready and data is copied over locally, send to CVAT for manual inspection/correction

- [x] Convert annotations to Ultralytics YOLO format
- [x] reduce image size (note: will need change in bbox coordinates)
- [x] map labels -> all classes to plant, colorchecker, non-target weeds
- [x] create structured data zip (to upload)
- [x] upload data to project (new project or new task in the same project over time) using cli/sdk
- [x] Initial setup script - get cvat image, create user, etc.
- [x] test setup on sunny

Needs manual intervention on CVAT. These will move to a different mode (get data on LTS, check that location for updated annotations, etc)
- [ ] ~relabel + export cvat results~
- [ ] ~Apply cvat results to images (note: image would be resized - coordinates should be normalized)~

### PR DESCRIPTION
**Pull request by jinamshah**
Fixes #7 
This pull request introduces significant changes to the configuration, dataset processing, and CVAT integration functionalities. The most important changes include the addition of new tasks and configurations in the `config.yaml` file, the creation of new scripts for CVAT setup and integration, and the implementation of new classes for formatting and importing datasets for CVAT.

### Configuration Changes:
* [`conf/config.yaml`](diffhunk://#diff-cc6ff10f70311db9bd57e102b0f7a2258df3266a18837e4489daa366210cb31bL10-R28): Added new tasks `cvat_formatter` and `cvat_importer`, and new configurations for CVAT including output path, default image dimensions, resize factor, class mapping, and dataset format.

### Dataset Processing:
* [`src/cvat_formatter.py`](diffhunk://#diff-93e8c07fc19e9d324ee0f821142f9aadcdf5040526a62af89c0bd0278fc665ceR1-R323): Implemented the `CVATFormatter` class to format images and annotations in the Ultralytics YOLO Detection format for CVAT, including methods for loading datasets, converting bounding boxes, processing images, and creating necessary files for CVAT import.
* [`src/cvat_importer.py`](diffhunk://#diff-b00329accbf140dc9de08f36d08aca591d54bea52110887f1e2dd73dbc3f3844R1-R73): Added the `CVATImporter` class to handle the uploading of datasets to the CVAT server, including methods for logging in, creating projects, and uploading datasets.

### CVAT Integration:
* [`setup_cvat.sh`](diffhunk://#diff-26e82d82906fef3f1d1d14b79331038c143b8e10c650cbe1865bc1d20e5ebd8fR1-R18): Added a script for setting up CVAT on the server, including cloning the CVAT repository, setting up Docker, and creating a superuser.
* [`environment.yaml`](diffhunk://#diff-0ff3348ae0de662ef053d7de057a7a92c091a09fc3d245c8da02bec028ca3d0bL13-R14): Added `cvat-sdk` and `opencv` to the dependencies to support CVAT integration and image processing.

### Utility Functions:
* [`src/utils/utils.py`](diffhunk://#diff-ea2db586282a8ad1000497fab6d8d11e91c7988ada13e74d86ab780f4010af3bR1-R49): Added utility functions `read_secrets` for reading secrets from a YAML file and `find_most_recent_dataset_path` for finding the most recent training dataset directory based on a hierarchical date/time structure.

### CODE REVIEW

**Comment by mkutu** [comment]
@jinamshah will pull and merge when this is ready for review

File: src/cvat_importer.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        self.cvat_client.login(credentials=[self.cvat_secrets['username'], self.cvat_secrets['password']])
+    
+    def create_project(self) -> str:
+        """
+        Create a new CVAT project using the most recent dataset
+        """
+        experiment_date_path = find_most_recent_dataset_path(self.cfg.database.dataset.output_path)
+        
+        # Extract date and time components from dataset path
+        path_parts = str(experiment_date_path).split('/')
```
[the remark this answers is not in the document]
Comment (jinamshah) [reply src/cvat_importer.py]: fixed

File: src/download_images.py [hunk 2]
```diff
@@ -26,21 +26,35 @@ def __init__(self, cfg: DictConfig) -> None:
...
         else:
             self.dataset_path = base_dataset_path
             log.warning(f"Dataset path does not exist: {self.dataset_path}")
+            
+            if date_dirs:
+                self.dataset_path = date_dirs[0]
+                log.info(f"Using most recent dataset directory: {self.dataset_path}")
+            else:
+                self.dataset_path = base_dataset_path
+                log.warning(f"No timestamp directories found, using base path: {self.dataset_path}")
```
[the remark this answers is not in the document]
Comment (jinamshah) [reply src/download_images.py]: using utils func now
